src/ColumnSchema.php

```diff
@@ -72,32 +71,24 @@ final class ColumnSchema extends AbstractColumnSchema
      */
     public function dbTypecast(mixed $value): mixed
     {
-        if ($value === null) {
-            return null;
-        }
-
-        if ($value instanceof ExpressionInterface) {
+        if ($value === null || $value instanceof ExpressionInterface) {
             return $value;
         }
 
         if ($this->dimension > 0) {
             return new ArrayExpression($value, $this->getDbType(), $this->dimension);
         }
 
-        if (in_array($this->getDbType(), [SchemaInterface::TYPE_JSON, SchemaInterface::TYPE_JSONB], true)) {
-            return new JsonExpression($value, $this->getDbType());
-        }
-
-        if (is_string($value) && $this->getType() === SchemaInterface::TYPE_BINARY) {
-            /** explicitly setup PDO param type for binary column */
-            return new Param($value, PDO::PARAM_LOB);
-        }
-
-        if (is_int($value) && $this->getType() === Schema::TYPE_BIT) {
-            return str_pad(decbin($value), $this->getSize() ?? 0, '0', STR_PAD_LEFT);
-        }
-
-        return $this->typecast($value);
+        return match ($this->getType()) {
+            SchemaInterface::TYPE_JSON => new JsonExpression($value, $this->getDbType()),
```

Comment:
check getType not equal check getDbType
// if (in_array($this->getDbType(), [SchemaInterface::TYPE_JSON, SchemaInterface::TYPE_JSONB], true)) {

Reply:
Need add test for this behaviour

Reply:
Test added